src/mds/Server.cc

```diff
@@ -3108,6 +3117,9 @@ CInode* Server::prepare_new_inode(MDRequestRef& mdr, CDir *dir, inodeno_t useino
 	     << " (" << mdr->session->info.prealloc_inos
 	     << ", " << mdr->session->info.prealloc_inos.size() << " left)"
 	     << dendl;
+  } else if (!mdr->session && mdr->client_request->is_queued_for_replay()) {
+    mdr->alloc_ino = 
+      in->inode.ino = mds->inotable->project_alloc_id(useino);
```

Comment:
need to check if useino is inotable. If not,  return err. assume request is completed

Reply:
changed